Mathlib/Data/Finset/Prod.lean

```diff
@@ -428,6 +428,16 @@ theorem offDiag_insert (has : a ∉ s) : (insert a s).offDiag = s.offDiag ∪ {a
     union_empty, union_right_comm]
 #align finset.off_diag_insert Finset.offDiag_insert
 
+theorem offDiag_filter_lt_eq_filter_le {ι}
+    [PartialOrder ι] [DecidableEq ι]
+    [DecidableRel (LE.le (α := ι))] [DecidableRel (LT.lt (α := ι))]
+    (s : Finset ι) :
+    s.offDiag.filter (fun i => i.1 < i.2) = s.offDiag.filter (fun i => i.1 ≤ i.2) := by
+  ext ⟨i, j⟩
+  simp only [mem_filter, mem_offDiag, ne_eq, and_congr_right_iff, and_imp]
+  rintro _ _ h
+  rw [Ne.le_iff_lt h]
```

Comment:
Can you use the newly introduced `Finset.filter_inj` from #13827 ?

Reply:
That's about the sets being equal, not the conditions? I can add a primed lemma though.